rfcs/0001-multi-tenancy/README.md

```diff
@@ -0,0 +1,166 @@
+for more details on how to harden shared clusters.
+
+##### Caveats
+
+Currently, Flux does not enforce a service account to be specified on Flux `Kustomizations` and `HelmReleases`.
+When a service account is not specified, Flux defaults to cluster-admin.
+In order to enforce the tenant isolation, an admission controller such as Kyverno or OPA Gatekeeper must be used
+to make the `.spec.serviceAccountName` a required field for the Flux custom resources created by tenants.
+
+We provide an [example](https://github.com/fluxcd/flux2-multi-tenancy/blob/main/infrastructure/kyverno-policies/flux-multi-tenancy.yaml) 
+for enforcing service accounts using a Kyverno cluster policy.
+
+Currently, Flux allows for `Kustomizations` and `HelmReleases` to reference sources
+(`GitRepositories`, `HelmRepositories` and `Buckets`) across namespaces.
+In order to prevent tenants from accessing each other sources, an admission controller such as Kyverno or OPA Gatekeeper
+must be used to block cross-namespace references.
+
+We provide an [example](https://github.com/fluxcd/flux2-multi-tenancy/blob/main/infrastructure/kyverno-policies/flux-multi-tenancy.yaml)
+for blocking source cross-namespace references using a Kyverno cluster policy.
+
+#### Hard Multi-Tenancy
+
+With hard multi-tenancy, the platform admins use Kubernetes Cluster API to create dedicated clusters for each tenant.
+The Flux instance installed on the management cluster is responsible
+for reconciling the cluster definitions belonging to tenants.
+
+To enable GitOps for the tenant's clusters, the platform admins can configure the Flux instance running on the
+management cluster to connect to the tenant's cluster using the `kubeConfig` generated by the Cluster API provider.
+
+To configure Flux reconciliation of remote clusters, a Kubernetes secret containing a `kubeConfig` can be specified
+in Flux `Kustomizations` and `HelmReleases` under `.spec.kubeConfig.secretRef`. Please consult the Flux API
+documentation for more details:
+
+- [Kustomization API: Remote Clusters](https://fluxcd.io/docs/components/kustomize/kustomization/#remote-clusters--cluster-api)
+- [HelmRelease API: Remote Clusters](https://fluxcd.io/docs/components/helm/helmreleases/#remote-clusters--cluster-api)
+
+Note that with hard multi-tenancy, tenants have full access to cluster-wide resources, so they have the option
+to manage Flux independently of platform admins, by deploying a Flux instances on each cluster.
+
+##### Caveats
+
+When using a Kubernetes Cluster API provider, the `kubeConfig` secret is automatically generated and Flux can
+make use of it without any manual actions. For clusters created by other means than Cluster API, the
+platform team has to create the `kubeConfig` secrets to allow Flux access to the remote clusters.
+
+Currently, we don't provide any guidance for cluster admins on how to generate the `kubeConfig` secrets.
```

Comment:
`Currently` is used a couple of times in the document. But I think placing it in context of date and time (and maintaining this) later on will not be easy. Would it be possible to avoid "point in time" references, and maybe place the document as a whole in a moment in time in the summary?

Reply:
Maybe "As of Flux v0.21.0" instead of "Current"?

Reply:
Yes, that would already be better I think.

Reply:
I'm now mentioning the version instead of "current" in all opened RFCs.